src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -1,11 +1,12 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T extends Machine> {
     /**
      * In your implementations - create 2-3 models, add them to the list and return
      * @return - the list of models
      */
-    List<Object> get();
+    List<Machine> get();
```

Comment:
```suggestion
    List<T> get();
```

Reply:
> Suggested change
>     List<Machine> get();
>     List<T> get();

If I removed it, override method get()in producers classes will return List of producer type. But in MashineServiceImpl in getAll() method we need to recieve list of Machine

Reply:
done